### DISCUSSION TITLE
custom search filters in a screen

### DISCUSSION THREAD

**Discussion by shah180** [orchidsoftware/platform#1597] [Q&A]
Hi,

I have created a product screen where i am  displaying the list of products in a table. Each column is filterable and it is working very well. I can filter by productname, and other columns (the filter icon on the column name). I can do sorting and all kind of stuff

Now I wanted to put a separate search area where I can filter by date range and other fields not present in the table. However, I dont see any example of how to do that (the column filter i have already done). Can someone please guide how can I create search area and filters on the top of the data table

Regards

**Comment by tabuna** [comment]
Hi, I don't have a step-by-step tutorial on how to do this right now. But there is an example that can be found in the user's browsing list. Right here [RoleFilter](https://github.com/orchidsoftware/platform/blob/cb2e41ef5fabdb57af17cf28913834236b523121/stubs/app/Orchid/Filters/RoleFilter.php#L1). And also the documentation section about this: https://orchid.software/en/docs/filters/#eloquent-filter

**Reply by Enakoneschniy** [reply]
@tabuna Hi, only RoleFilter works. I created a new filter and implemented it as a RoleFilter, but the "run" method won't run.
